bigtable/bigtable.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/bash
+#    Copyright 2018 Google, Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# This actions installs cloud-bigtable-client (https://github.com/GoogleCloudPlatform/cloud-bigtable-client/)
+# on dataproc cluster and configure it to use cloud BigTable (https://cloud.google.com/bigtable/).
+
+set -euxo pipefail
+
+readonly HBASE_HOME='/usr/lib/hbase'
+readonly BIGTABLE_HBASE_CLIENT='bigtable-hbase-1.x-hadoop-1.3.0.jar'
+readonly HBASE_BIGTABLE_DL_LINK="http://central.maven.org/maven2/com/google/cloud/bigtable/bigtable-hbase-1.x-hadoop/1.3.0/${BIGTABLE_HBASE_CLIENT}"
+
+function update_apt_get() {
+  for ((i = 0; i < 10; i++)); do
+    if apt-get update; then
+      return 0
+    fi
+    sleep 5
+  done
+  return 1
+}
+
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
+  return 1
+}
+
+function install_big_table_client() {
+    wget -q "${HBASE_BIGTABLE_DL_LINK}" -O "${HBASE_HOME}/lib/${BIGTABLE_HBASE_CLIENT}" \
+      || err 'Unable to install BigTable client libs.'
+    ln -s "${HBASE_HOME}/lib/${BIGTABLE_HBASE_CLIENT}" \
+      "/usr/lib/hadoop-mapreduce/lib/${BIGTABLE_HBASE_CLIENT}"
+    echo "export PATH=\"${HBASE_HOME}/bin:$PATH\"" >> /etc/profile
+}
+
+function configure_big_table_client() {
+
+#Update classpaths
+ cat << 'EOF' >> /etc/hadoop/conf/hadoop-env.sh
+HADOOP_CLASSPATH="${HADOOP_CLASSPATH}:/usr/lib/hbase/*"
+HADOOP_CLASSPATH="${HADOOP_CLASSPATH}:/usr/lib/hbase/lib/*"
+HADOOP_CLASSPATH="${HADOOP_CLASSPATH}:/etc/hbase/conf"
+EOF
+  cat << 'EOF' >> /etc/hadoop/conf/mapred-env.sh
+HADOOP_CLASSPATH="${HADOOP_CLASSPATH}:/usr/lib/hbase/*"
+HADOOP_CLASSPATH="${HADOOP_CLASSPATH}:/usr/lib/hbase/lib/*"
+HADOOP_CLASSPATH="${HADOOP_CLASSPATH}:/etc/hbase/conf"
+EOF
```

Comment:
Don't edit both hadoop-env and mapred-env.

I would recommend just mapred-env

Reply:
Without modyfication of `hadoop-env` submitting wordcount-hbase job from the examples provided in README.md results in error:
>java.lang.NoClassDefFoundError: org/apache/hadoop/hbase/mapreduce/Export
>	at com.example.bigtable.sample.WordCountDriver.main(WordCountDriver.java:33)

The command : `hadoop classpath` doesn't return HBase in libraries. I will take closer look at that.